internal/cmd/context/ssh_key.go

```diff
@@ -0,0 +1,211 @@
+package context
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/spf13/cobra"
+
+	"github.com/hetznercloud/cli/internal/cmd/cmpl"
+	"github.com/hetznercloud/cli/internal/hcapi2"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+// TODO: make *state.Config mockable and write tests
+
+func newSSHKeyCommand(s state.State) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "ssh-key",
+		Short:                 "Manage a context's default SSH key",
+		Args:                  cobra.NoArgs,
+		TraverseChildren:      true,
+		DisableFlagsInUseLine: true,
+	}
+	cmd.AddCommand(newSSHKeyAddCommand(s))
+	cmd.AddCommand(newSSHKeyRemoveCommand(s))
+	cmd.AddCommand(newSSHKeyListCommand(s))
+	return cmd
+}
+
+func newSSHKeyAddCommand(s state.State) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "add SSH-KEY...",
```

Comment:
I feel this breaks the behaviour of `hcloud context`. All other subcommands take the name of the context as a positional argument, while this one takes the ssh-key as positional and context as flag.

Reply:
You can still make it an optional positional arg, defaulting to the current index.

I checked some other commands and they all have the pattern `hcloud <resource> <action> <resource-name-or-id>`. The commands matching our current pattern would be:

- `hcloud context add-ssh-keys [context] --ssh-key foobar`
- `hcloud context remove-ssh-keys [context] --ssh-key foobar`
- `hcloud context list-ssh-keys [context] --ssh-key foobar`

I just noticed that you actually added another level to the command, so the pattern mentioned above actually matches the last three segments. And I am kinda fine with it.

@jooola any opinions?

Reply:
No strong opinion, but I prefer less nested commands, discovery is easier when one need to read the help pages.

About the flags, I would follow the convention we already use in the CLI.

Using `hcloud context remove-ssh-keys [context] --ssh-key foobar`

Reply:
I think having a dedicated subcommand helps with sorting and provides a better structure.
With the proposed subcommands the help would look like this:
```
$ hcloud context -h
Manage contexts

Usage:
  hcloud context [command]

Available Commands:
  active      Show active context
  add-ssh-keys ...
  create      Create a new context
  delete      Delete a context
  list        List contexts
  list-ssh-keys ...
  remove-ssh-keys ...
  use         Use a context
```

Reply:
Is it possible to order the command manually?

Reply:
We can use cobra `Groups` to group the output into various ... groups. I would love to see that anyway for `hcloud server` and other huge commands.

Reply:
I still think having a dedicated subcommand for it is nice because command names are kept shorter and the `hcloud context ssh-key` command follows the rough structure of other subcommands (add/remove/list).
Yes, I agree that we should group subcommands for larger commands.